test/e2e_tests/data_validation_test.go

```diff
@@ -0,0 +1,201 @@
+		err := os.RemoveAll(path)
+		suite.Equal(nil, err)
+	}
+}
+
+func (suite *dataValidationTestSuite) validateData(localFilePath string, remoteFilePath string) {
+	// copy to mounted directory
+	cpCmd := exec.Command("cp", localFilePath, remoteFilePath)
+	cliOut, err := cpCmd.Output()
+	fmt.Println(string(cliOut))
+	suite.Equal(nil, err)
+
+	// delete the cache directory
+	suite.dataValidationTestCleanup([]string{suite.testCachePath})
+
+	// compare the local and mounted files
+	diffCmd := exec.Command("diff", localFilePath, remoteFilePath)
+	cliOut, err = diffCmd.Output()
+	fmt.Println(string(cliOut))
+	suite.Equal(0, len(cliOut))
+	suite.Equal(nil, err)
+}
+
+// -------------- Data Validation Tests -------------------
+
+// data validation for small sized files
+func (suite *dataValidationTestSuite) TestSmallFileData() {
+	fileName := "small_data.txt"
+	localFilePath := suite.testLocalPath + "/" + fileName
+	remoteFilePath := suite.testMntPath + "/" + fileName
+
+	// create the file in local directory
+	srcFile, err := os.OpenFile(localFilePath, os.O_CREATE, 0777)
+	suite.Equal(nil, err)
+	srcFile.Close()
+
+	// write to file in the local directory
+	err = ioutil.WriteFile(localFilePath, suite.minBuff, 0777)
+	suite.Equal(nil, err)
+
+	suite.validateData(localFilePath, remoteFilePath)
+
+	suite.dataValidationTestCleanup([]string{localFilePath, remoteFilePath, suite.testCachePath})
+}
+
+// data validation for medium sized files
+func (suite *dataValidationTestSuite) TestMediumFileData() {
+	fileName := "medium_data.txt"
+	localFilePath := suite.testLocalPath + "/" + fileName
+	remoteFilePath := suite.testMntPath + "/" + fileName
+
+	// create the file in local directory
+	srcFile, err := os.OpenFile(localFilePath, os.O_CREATE, 0777)
+	suite.Equal(nil, err)
+	srcFile.Close()
+
+	// write to file in the local directory
+	err = ioutil.WriteFile(localFilePath, suite.medBuff, 0777)
+	suite.Equal(nil, err)
+
+	suite.validateData(localFilePath, remoteFilePath)
+
+	suite.dataValidationTestCleanup([]string{localFilePath, remoteFilePath, suite.testCachePath})
+}
+
+// data validation for large sized files
+func (suite *dataValidationTestSuite) TestLargeFileData() {
+	fileName := "large_data.txt"
+	localFilePath := suite.testLocalPath + "/" + fileName
+	remoteFilePath := suite.testMntPath + "/" + fileName
+
+	// create the file in local directory
+	srcFile, err := os.OpenFile(localFilePath, os.O_CREATE, 0777)
+	suite.Equal(nil, err)
+	srcFile.Close()
+
+	// write to file in the local directory
+	err = ioutil.WriteFile(localFilePath, suite.hugeBuff, 0777)
+	suite.Equal(nil, err)
+
+	suite.validateData(localFilePath, remoteFilePath)
+
+	suite.dataValidationTestCleanup([]string{localFilePath, remoteFilePath, suite.testCachePath})
+}
+
+// -------------- Main Method -------------------
+func TestDataValidationTestSuite(t *testing.T) {
+	initDataValidationFlags()
+	dataValidationTest := dataValidationTestSuite{
+		minBuff:  make([]byte, 1024),
```

Comment:
Instead of having just one file upload/download of each size, lets enhance this e2e test to do multiple such files in parallel e.g. 10 files of each size to be uploaded/downloaded in parallel.

Reply:
Will take this up in separate PR